### ISSUE TITLE
repeat.for throws error when wrapped with if.bind

### ISSUE THREAD

**Issue by bravius** [aurelia/aurelia#1119]
# 🐛 Bug Report

When a repeat.for element is wrapped with an element containing if.bind, it throws "Uncaught Error".

```
<div if.bind="items.length>0">
  <div repeat.for="item of items">
    ...
  </div>
</div>

```
## 🤔 Expected Behavior

Shouldn't throw error... lol.

## 😯 Current Behavior

Throws the following error:

Uncaught Error: viewsLen=2, mapLen=1
    at Repeat.createAndActivateAndSortViewsByKey (VM157 entry-bundle.js:23859)
    at VM157 entry-bundle.js:23681
    at onResolve (VM157 entry-bundle.js:3175)
    at Repeat.handleCollectionChange (VM157 entry-bundle.js:23679)
    at SubscriberRecord.notifyCollection (VM157 entry-bundle.js:37853)
    at ArrayObserver.notify (VM157 entry-bundle.js:34993)
    at Array.splice (VM157 entry-bundle.js:34873)
    at App.remove (VM157 entry-bundle.js:195)
    at CallScopeExpression.evaluate (VM157 entry-bundle.js:31010)
    at Listener.callSource (VM157 entry-bundle.js:15425)

## 🔦 Context

Using latest aurelia@dev (0.10.0-dev.202102161145)

## 💻 Code Sample

See gist to replicate the bug.  

Click "add" to add some list items, then "remove" on each one to empty the list. 

Repeating the add/remove procedure will throw "Uncaught Error: viewsLen=2, mapLen=1"

https://gist.dumber.app/?gist=a29951a42030975f2fdcce78b42de6b1

Note: if the wrapping element uses show.bind instead of if.bind, no error is thrown

**Comment by bigopon** [comment]
@bravius thank you for creating this issue! This is because `[if]` running before `[repeat]` causing `[repeat]` to skip removing existing views when the value of `[if]` evaluates to false. A fix should be in soon.

**Comment by bigopon** [comment]
Closed by #1511
